iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -411,59 +409,57 @@ def createViewer(self, scriptUrl: str = DEFAULT_VIEWER_PATH, useStubViewer: bool
             if report.modelDocument.type == Type.INLINEXBRLDOCUMENTSET:
                 # Sort by object index to preserve order in which files were specified.
                 xmlDocsByFilename = {
-                    os.path.basename(self.outputFilename(doc.filepath)): deepcopy(doc.xmlDocument)
+                    os.path.basename(self.outputFilename(doc.filepath)): doc.xmlDocument
                     for doc in sorted(report.modelDocument.referencesDocument.keys(), key=lambda x: x.objectIndex)
                     if doc.type == Type.INLINEXBRL
                 }
                 docSetFiles = list(xmlDocsByFilename.keys())
 
-                if xmlDocument is None:
-                    if useStubViewer:
-                        xmlDocument = self.getStubDocument()
-                        iv.addFile(iXBRLViewerFile(DEFAULT_OUTPUT_NAME, xmlDocument))
-                    else:
-                        xmlDocument = next(iter(xmlDocsByFilename.values()))
-
                 for filename, docSetXMLDoc in xmlDocsByFilename.items():
                     iv.addFile(iXBRLViewerFile(filename, docSetXMLDoc))
 
             elif useStubViewer:
-                if xmlDocument is None:
-                    xmlDocument = self.getStubDocument()
-                    iv.addFile(iXBRLViewerFile(DEFAULT_OUTPUT_NAME, xmlDocument))
                 filename = self.outputFilename(os.path.basename(report.modelDocument.filepath))
                 docSetFiles = [ filename ]
                 iv.addFile(iXBRLViewerFile(filename, report.modelDocument.xmlDocument))
 
             else:
+                srcFilename = self.outputFilename(os.path.basename(report.modelDocument.filepath))
+                docSetFiles = [ srcFilename ]
                 if len(self.reports) == 1:
                     # If there is only a single report, call the output file "xbrlviewer.html"
                     filename = "xbrlviewer.html"
                 else:
                     # Otherwise, preserve filenames
-                    filename = self.outputFilename(os.path.basename(report.modelDocument.filepath))
-                    docSetFiles = [ filename ]
-                if xmlDocument is None:
-                    xmlDocument = deepcopy(report.modelDocument.xmlDocument)
-                    iv.addFile(iXBRLViewerFile(filename, xmlDocument))
-                else:
-                    iv.addFile(iXBRLViewerFile(filename, report.modelDocument.xmlDocument))
+                    filename = srcFilename
+                iv.addFile(iXBRLViewerFile(filename, report.modelDocument.xmlDocument))
+
+            docSetKey = frozenset(docSetFiles)
+            sourceReport = sourceReportsByFiles.get(docSetKey)
+            if sourceReport is None:
+                sourceReport = self.addSourceReport()
+                sourceReportsByFiles[docSetKey] = sourceReport
+                sourceReport["docSetFiles"] = list(urllib.parse.quote(f) for f in docSetFiles)
```

Comment:
Does this change the format of the JSON (previously plain strings, now URL-quoted strings)?  That seems unfortunate.  It also seems generally odd to store pre-quoted strings in the JSON.

Reply:
This change was introduced here: https://github.com/Arelle/ixbrl-viewer/pull/394#discussion_r1151188258

The rationale is that `docSetFiles` should be a list of URLs rather than files, hence URL quoting here.

Reply:
I did not see the pre-existing `urllib` call, but I see it now, thanks.  I also had not remembered that we made that change, and I guess it was backwards incompatible then too...  Anyway, perhaps it would be helpful to update [the doc](https://github.com/Arelle/ixbrl-viewer/blob/master/ARCHITECTURE.md#docsetfiles) to say it's a list of URLs rather than files.

Reply:
I think it was more of a bug fix that a backwards incompatible change, as the viewer code was always treating them as URLs, so if you previously tried to use a filename with a URL-significant character such as `?` or `#` in it, it would break.

I've updated the docs, and in doing so, also remove the note about not using `docSetFiles` for single document IXDS, as this PR also removes that historical anomaly.

Reply:
Alternatively, it could have been considered a bug in the JS.  That way, generated viewers are fixed by just swapping in the new JS rather than needing to be regenerated.  Anyway, thanks for the docs updates.